kafka-connect-hive/src/main/scala/com/landoop/streamreactor/connect/hive/AsyncFunctionLoop.scala

```diff
@@ -0,0 +1,48 @@
+package com.landoop.streamreactor.connect.hive
+
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.scalalogging.slf4j.StrictLogging
+
+import scala.concurrent.duration.Duration
+
+class AsyncFunctionLoop(interval: Duration, description: String)(thunk: => Unit)
+  extends AutoCloseable
+    with StrictLogging {
+
+  @volatile private var running = false
```

Comment:
In general I think they recommend you use Atomic's rather than @volatiles, but this isn't incorrect.